### ISSUE TITLE
Global Autotype does not match anymore

### ISSUE THREAD

**Issue by benwalch** [MacPass/MacPass#922]
Global Autotype regex matching stopped working.
We run many instances of a CMS and the credentials of the users are synchronized by a central service. So we have the same credentials for every instance of this CMS. We make use of the RegEx match of Autotype Window-Association to have one entry to match all our CMS-Logins:
Autotype Window Association of the Entry: ```CMS Login:(.*)```
which matched ```CMS Login: Customer A```, ```CMS Login: Customer B```, etc.
Since Version 0.7.4 this behaviour seems to be broken and does not work anymore.

**To Reproduce**
Use MacPass > v.0.7.3
Create an Entry and create an Autotype Window Association: ```CMS Login:(.*)```

Open a Page in a Browser with the title: ```CMS Login: SomeCool String``` and press the global autotype sequence.

**Expected behavior**
Autotype matches and puts in the credentials of the corresponding entry.

**Desktop**
 - macOS: 10.13.6
 - Version: > 0.7.3

**Comment by mstarke** [comment]
This is as intended. 0.7.3 and before were not strict on the matching and did some accidental reg-exp expansion which should not have occured. Your example hits exactly this spot. Either you declare the window title as an reg-ex by enclosing it with forward slashes ```//CMS Login:(.*)//``` or you change the match to a simple wildcard which should be enough for your use-case ```CMS Login:*```

Also consider that your regex will also match string that do end with ```CMS Login:``` unless you add an anchor ```^``` but the wildcard match will only match strings that start with ```CMS Login:```

**Comment by mstarke** [comment]
To clarify the change in more detail. This change is in KeePassKit 2.3.8 which was included in MacPass 0.7.4 (as KeePassKit 2.4.1).

```diff
+ static NSString * const _KPKWindowAssociationWildcard = @"KPKWindowAssociationWildcard";
- NSString *regEx = [self.windowTitle stringByReplacingOccurrencesOfString:@"*" withString:@".*" options:NSCaseInsensitiveSearch range:NSMakeRange(0, self.windowTitle.length)];
- /* ensure that the regex matches exactly */
+ NSString *regEx = [self.windowTitle stringByReplacingOccurrencesOfString:@"*" withString:_KPKWindowAssociationWildcard options:NSCaseInsensitiveSearch range:NSMakeRange(0, self.windowTitle.length)];
+ regEx = [NSRegularExpression escapedPatternForString:regEx];
+ regEx = [regEx stringByReplacingOccurrencesOfString:_KPKWindowAssociationWildcard withString:@".*"];
pattern = [NSString stringWithFormat:@"^%@$", regEx];
```

Based on your example:

0.7.3:

    Original: CMS Login:(.*)
    Expand Wildcard as Reg-Exp: CMS Login:(..*)

0.7.4:

    Original: CMS Login:(.*)
    Mask Wildcard: CMS Login:(.KPKWindowAssociationWildcard)
    Escape Regex: CMS Login:\(\.KPKWindowAssociationWildcard\)
    Unmask Wildcard: CMS Login:\(\..*\)

Which makes it pretty clear why in 0.7.3 it did work and in 0.7.4 not anymore. I'm sorry for the inconvenience in introducing this breaking change but I hope you understand my reasoning.
